convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -1411,21 +1411,90 @@ def test_call_yum_cmd_w_downgrades(monkeypatch, retcode, output):
-        ("2:8.2.3568-1.fc35", "2:8.2.3568-1.fc35", 0),
+        ("kernel-core-2:8.2.3568-1.fc35", "kernel-core-2:8.2.3568-1.fc35", 0),
     ),
 )
 def test_compare_package_versions(version1, version2, expected):
     assert pkghandler.compare_package_versions(version1, version2) == expected
 
 
+@pytest.mark.parametrize(
+    ("version1", "version2", "exception_message"),
+    (
+        ("kernel-core-0:390-287.fc36", "kernel-0:390-287.fc36", "kernel-core does not match kernel"),
+        ("kernel-core-0:390-287.fc36.aarch64", "kernel-core-0:391-287.fc36.i86", "aarch64 does not match i86"),
+    ),
+)
+def test_compare_package_versions_warnings(version1, version2, exception_message):
+
+    with pytest.raises(ValueError, match=exception_message):
+        pkghandler.compare_package_versions(version1, version2)
+
+
+def test_parse_pkg_string_dnf(monkeypatch):
+
+    pkg = "kernel-core-0:390-287.fc36.aarch64"
+    parse_pkg_with_dnf_mock = mock.Mock()
+    monkeypatch.setattr(pkghandler, "_parse_pkg_with_dnf", value=parse_pkg_with_dnf_mock)
+    monkeypatch.setattr(pkgmanager, "TYPE", "dnf")
+    pkghandler.parse_pkg_string(pkg)
+    parse_pkg_with_dnf_mock.assert_called_once()
+
+
+def test_parse_pkg_string_yum(monkeypatch):
+
+    pkg = "kernel-core-0:390-287.fc36.aarch64"
+    parse_pkg_with_yum_mock = mock.Mock()
+    monkeypatch.setattr(pkghandler, "_parse_pkg_with_yum", value=parse_pkg_with_yum_mock)
+    monkeypatch.setattr(pkgmanager, "TYPE", "yum")
+    pkghandler.parse_pkg_string(pkg)
+    parse_pkg_with_yum_mock.assert_called_once()
+
+
+# format: name, epoch, version, release, arch
+test_packages = (
+    ("kernel-core-0:4.18.0-240.10.1.el8_3.i86", ["kernel-core", "0", "4.18.0", "240.10.1.el8_3", "i86"]),  # NEVRA
+    ("kernel-core-0:4.18.0-240.10.1.el8_3", ["kernel-core", "0", "4.18.0", "240.10.1.el8_3", "None"]),  # NEVR
+    (
+        "1:NetworkManager-1.18.8-2.0.1.el7_9.aarch64",
+        ["NetworkManager", "1", "1.18.8", "2.0.1.el7_9", "aarch64"],
+    ),  # ENVRA
+    ("1:NetworkManager-1.18.8-2.0.1.el7_9", ["NetworkManager", "1", "1.18.8", "2.0.1.el7_9", "None"]),  # ENVR
+    (
+        "NetworkManager-1.18.8-2.0.1.el7_9.aarch64",
+        ["NetworkManager", "None", "1.18.8", "2.0.1.el7_9", "aarch64"],
+    ),  # NVRA
+    ("NetworkManager-1.18.8-2.0.1.el7_9", ["NetworkManager", "None", "1.18.8", "2.0.1.el7_9", "None"]),  # NVR
+)
+
+
+@pytest.mark.skipif(pkgmanager.TYPE == "dnf", reason="cannot test yum backend if yum is not present")
+@pytest.mark.parametrize(
+    ("package", "expected"),
+    (test_packages),
+)
+def test_parse_pkg_with_yum(package, expected):
+    assert pkghandler._parse_pkg_with_yum(package) == expected
+
+
+@pytest.mark.skipif(pkgmanager.TYPE == "yum", reason="cannot test dnf backend if dnf is not present")
+@pytest.mark.parametrize(
+    ("package", "expected"),
+    (test_packages),
+)
+def test_parse_pkg_with_dnf(package, expected):
+    assert pkghandler._parse_pkg_with_dnf(package) == expected
```

Comment:
You can transform that variable to be in all caps. Another thing is, I don't think this name is descriptive enough for what it's holding... But still, I can't figure out what name would suit it better here. 

```suggestion
TEST_PACKAGES = (
    ("kernel-core-0:4.18.0-240.10.1.el8_3.i86", ["kernel-core", "0", "4.18.0", "240.10.1.el8_3", "i86"]),  # NEVRA
    ("kernel-core-0:4.18.0-240.10.1.el8_3", ["kernel-core", "0", "4.18.0", "240.10.1.el8_3", "None"]),  # NEVR
    (
        "1:NetworkManager-1.18.8-2.0.1.el7_9.aarch64",
        ["NetworkManager", "1", "1.18.8", "2.0.1.el7_9", "aarch64"],
    ),  # ENVRA
    ("1:NetworkManager-1.18.8-2.0.1.el7_9", ["NetworkManager", "1", "1.18.8", "2.0.1.el7_9", "None"]),  # ENVR
    (
        "NetworkManager-1.18.8-2.0.1.el7_9.aarch64",
        ["NetworkManager", "None", "1.18.8", "2.0.1.el7_9", "aarch64"],
    ),  # NVRA
    ("NetworkManager-1.18.8-2.0.1.el7_9", ["NetworkManager", "None", "1.18.8", "2.0.1.el7_9", "None"]),  # NVR
)


@pytest.mark.skipif(pkgmanager.TYPE == "dnf", reason="cannot test yum backend if yum is not present")
@pytest.mark.parametrize(
    ("package", "expected"),
    (TEST_PACKAGES),
)
def test_parse_pkg_with_yum(package, expected):
    assert pkghandler._parse_pkg_with_yum(package) == expected


@pytest.mark.skipif(pkgmanager.TYPE == "yum", reason="cannot test dnf backend if dnf is not present")
@pytest.mark.parametrize(
    ("package", "expected"),
    (TEST_PACKAGES),
)
def test_parse_pkg_with_dnf(package, expected):
    assert pkghandler._parse_pkg_with_dnf(package) == expected
```